### DISCUSSION TITLE
STEP800の初期設定について

### DISCUSSION THREAD

**Discussion by arkwknsk** [ponoor/step-series-support#31] [Q&A] [answered]
STEP800をとても頼りにさせてもらっています

初期設定についてに質問させてください
microSDカードを差し込んでいない状態では、
電源を入れる度に、スピードや電圧などの各設定は
OSCで送信して設定する必要があるという認識でいいでしょうか？

**Comment by kanta** [comment]
STEP800を使っていただきありがとうございます。

おっしゃる通りで、各設定は電源が落ちると消えてしまい、次回電源投入時は、microSDを差し込んでいない状態では初期値に戻ってしまいます。ですので電源投入ごとにOSC経由で設定いただく必要があります。

**Reply by arkwknsk** [reply]
ご回答、ありがとうございます

長期展示の場合、microSDの寿命が気になるのですが
STEP800からはreadだけされている形でしょうか

**Reply by kanta** [reply, marked as answer]
そうですね、ファームウェア側の処理は起動時のファイル読み込みだけで、書き込みは行っていません。

OSC経由で設定する場合は、STEP800の起動時に `/booted` というメッセージが基板側から送られますので、これをきっかけに各種設定を送信するようにすれば、予期しない再起動が発生した場合などに対応できます。
https://ponoor.com/docs/step-series/osc-command-reference/automatically-sent-messages-from-step-400/

他の方法としては、ファームウェアの `loadConfig.cpp` というファイルに初期値が書き込まれていますので、これを変更してSTEP800本体に書き込んでしまう方法もあります。
https://github.com/ponoor/step-series-universal-firmware/blob/main/src/loadConfig.cpp

上位システムの状況に関わらず確実に読み込みができますが、将来本体を交換した際などにも同じファームウェア書き換え作業が必要ですので、長期展示のメンテナンス性という点では劣りそうです。

**Reply by arkwknsk** [reply]
@kanta 細かいアドバイス、ありがとうございます！参考にします
